Reorg to break dependency of `host` package on `enclave` package. Creation of two main methods.

joeldudley opened the pull request:
This does not include the work to create multiple workspaces, which will be part of a follow-up.

The two main methods - host and enclave - can connect to one another. However, they fall over once connected, as the host does not have an L1 node to connect to, and is also not able to communicate with any peers. I thought leaving it this way was preferable for now, rather than replicating the simulation by passing in various fake L1 nodes and peers. We can expand it to make it work later on.

tudor-malene commented on go/obscuronode/nodecommon/types.go:
```diff
@@ -4,17 +4,21 @@ import (
 	"github.com/ethereum/go-ethereum/common"
 )
 
 // Todo - this has to be a trie root eventually
```
this todo should be on the StateRoot type

joeldudley replied:
Addressed.

tudor-malene commented on go/obscuronode/nodecommon/enclave.go:
```diff
@@ -0,0 +1,79 @@
+	"github.com/obscuronet/obscuro-playground/go/obscurocommon"
+)
+
+// Enclave - The actual implementation of this interface will call an rpc service
```
this comment should be change as well, as it does that now

joeldudley replied:
Addressed.

tudor-malene reviewed (changes requested):
Thanks Joel.
I had a very quick look. Looks good in principle.
Can you please add a Readme.md in each package explaining what it does?

joeldudley commented:
@tudor-malene Addressed your comments and added readmes. The readmes are based on my best understanding and it would be good to validate them. I think there are various classes and functions that aren't in exactly the right place given the readme descriptions - we should probably fix this in the future.

tudor-malene commented on go/obscuronode/enclave/README.md:
```diff
@@ -0,0 +1 @@
+This package contains code related to the node's enclave component.
```
Umm. I think we should elaborate more here. 
Talk a bit how this has to be compiled with ego, started in a separate process, from the main method, etc

tudor-malene commented on go/obscuronode/nodecommon/README.md:
```diff
@@ -0,0 +1 @@
+This package contains code that is shared between the node host and the node enclave.
```
also maybe a bit more elaboreated.
This is basically the API to the enclave

otherview commented on go/obscuronode/enclave/main/main.go:
```diff
@@ -0,0 +1,57 @@
+	nodeAddressBytes, port := parseCLIArgs()
+
+	nodeAddress := common.BytesToAddress([]byte(*nodeAddressBytes))
+	if _, err := enclave.StartServer(*port, nodeAddress, nil); err != nil {
+		panic(err)
+	}
+
+	fmt.Printf("Enclave server listening on port %d.\n", *port)
+	select {}
```
What if there was a graceful stop mechanism? Would a chan that this select listens on, work ?

joeldudley replied:
But what would send to that chan? This is being interrupted by some external action (ctrl + c).

otherview reviewed (changes requested):
The refactor of the packages looks good. However I wouldn't include the obscuronode cmd and enclave cmd if they aren't working - Probably worth including those with the P2P PR instead.
